drivers/webdav/onitu_webdav/tests/driver.py

```diff
@@ -0,0 +1,76 @@
+import os
+import hashlib
+
+from io import BytesIO
+
+from tests.utils import driver
+from onitu_webdav.wd import get_WEBDAV_client, create_dirs
+from onitu.utils import get_random_string, b
+
+
+class Driver(driver.Driver):
+    def __init__(self, *args, **options):
+        options['hostname'] = os.getenv(
+            "ONITU_WEBDAV_HOSTNAME", "http://localhost"
+        )
+        options['username'] = os.getenv("ONITU_WEBDAV_USERNAME", "")
+        options['password'] = os.getenv("ONITU_WEBDAV_PASSWORD", "")
+        options['changes_timer'] = os.getenv("ONITU_WEBDAV_CHANGES_TIMER", 5)
+        root = os.getenv("ONITU_WEBDAV_ROOT", u"/")
+
+        self._root = root + get_random_string(10)
+
+        hostname = options['hostname']
+        username = options['username']
+        password = options['password']
+
+        super(Driver, self).__init__('webdav', *args, **options)
+
+        self.webd = get_WEBDAV_client(hostname, username, password)
+
+        create_dirs(self.webd, self._root)
+
+    @property
+    def root(self):
+        return self._root
+
+    def close(self):
+        self.rmdir(self.root)
+
+    def mkdir(self, subdirs):
+        create_dirs(self.webd, subdirs)
+
+    def rmdir(self, path):
+        self.webd.clean(b(path))
+
+    def write(self, filename, content):
+        create_dirs(self.webd, os.path.dirname(filename))
```

Comment:
This will return backslashes on Windows, so you might be better of using a simple split.